joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/controlflow/cfgcreation/CfgCreator.scala

```diff
@@ -536,6 +538,48 @@ class CfgCreator(entryNode: Method, diffGraph: DiffGraphBuilder) {
         )
     }
   }
+
+  /** The CFGs for match cases are modeled after PHP match expressions and assumes that a case will always consist of
+    * one or more JumpTargets followed by a single expression. The CFG also assumes an implicit `break` at the end of
+    * each match case.
+    */
+  protected def cfgsForMatchCases(body: AstNode): List[Cfg] = {
+    body.astChildren
+      .foldLeft(List(Cfg.empty)) { case (currCfg :: prevCfgs, astNode) =>
+        astNode match {
+          case jumpTarget: JumpTarget =>
+            val jumpCfg = cfgForJumpTarget(jumpTarget)
+            (currCfg ++ jumpCfg) :: prevCfgs
+
+          case node: AstNode =>
+            val nodeCfg = cfgFor(node)
+            Cfg.empty :: (currCfg ++ nodeCfg) :: prevCfgs
+        }
+      }
+      .reverse
+  }
+
+  /** CFG creation for match expressions of the form `match { case condition: expr ... }`
+    */
+  protected def cfgForMatchExpression(node: ControlStructure): Cfg = {
```

Comment:
Isn't that mostly like cfgForSwitchStatement?

Reply:
They are indeed very similar. I've refactored cfgForMatch/Switch to reduce code duplication.